sources/packages/backend/libs/integrations/src/cas/models/cas-service.model.ts

```diff
@@ -0,0 +1,108 @@
+import { CASSupplierRecordStatus, CASSupplierSiteStatus } from "@sims/sims-db";
+
+export class CASSupplierResponse {
+  items: CASSupplierResponseItem[];
+  hasMore: boolean;
+  limit: number;
+  offset: number;
+  count: number;
+}
+export class CASSupplierResponseItem {
+  suppliernumber: string;
+  suppliername: string;
+  subcategory: string;
+  sin: string;
+  providerid?: string;
+  businessnumber?: null;
+  status: CASSupplierRecordStatus;
+  supplierprotected?: "Y" | "N" | null;
+  standardindustryclassification?: string;
+  lastupdated: string;
+  supplieraddress: CASSupplierResponseItemAddress[];
+}
+
+export class CASSupplierResponseItemAddress {
+  suppliersitecode: string;
+  addressline1: string;
+  addressline2?: string;
+  addressline3?: string;
+  city: string;
+  province: string;
+  country: string;
+  postalcode: string;
+  emailaddress?: string;
+  accountnumber?: string;
+  branchnumber?: string;
+  banknumber?: string;
+  eftadvicepref?: string;
+  providerid?: string;
+  status: CASSupplierSiteStatus;
+  siteprotected?: string;
+  lastupdated: string;
+}
+
+export class CASAuthDetails {
+  access_token: string;
+  token_type: string;
+  expires_in: number;
+}
+
+/**
+ * Information needed for supplier and site creation on CAS.
+ */
+export class CreateSupplierAndSiteData {
+  firstName: string;
+  lastName: string;
+  sin: string;
+  emailAddress: string;
+  supplierSite: CreateSupplierSite;
+}
+
+/**
+ * Site information needed for supplier and site creation on CAS.
+ */
+export class CreateSupplierSite {
+  addressLine1: string;
+  city: string;
+  provinceCode: string;
+  postalCode: string;
+}
+
+/**
+ * Result from a supplier and site creation.
+ */
+export class CreateSupplierAndSiteResult {
+  supplierNumber: string;
+  supplierSiteCode: string;
+}
+
+/**
+ * Data used during the creation of a supplier and site on CAS.
+ * Some data transformation is needed to follow the CAS requirements.
+ * This payload is used to submit the data and also to be returned to
+ * the consumer, allowing it to be aware of the data actually submitted.
+ */
+export class CreateSupplierAndSiteSubmittedData {
+  SupplierName: string;
```

Comment:
Just for my understanding and knowledge, why do the keys for the object start with Uppercase? Any specific reason @andrewsignori-aot that they (CAS) might be doing this?

Reply:
I am not sure about the reason. It is not consistent in the docs either.
```json
{
  "Supplier_Number":"123456789",
  "Supplier_Site_Code":"001",
  "CAS-Returned-Messages":"SUCCESS"
}
```
vs
```json
{
  "SupplierName":"EXAMPLE, SUPPLIER 1",
  "SubCategory":"Individual",
  "Sin":"",
  "BusinessNumber":""
}
```
vs
```json
{
  "supplierprotected": null,
  "standardindustryclassification": null,
  "lastupdated": "2021-06-08 08:44:41",
}
```